src/components/LogMoodModal/LogMoodModal.tsx

```diff
@@ -1,34 +1,85 @@
 import iconClose from "../../assets/images/icon-close.svg";
-import { useState } from "react";
-import Stepper from "../Stepper/Stepper";
+import { useRef, useEffect, useState } from "react";
+import ModalContent from "./ModalContent";
 import MoodOptionList from "../MoodOptionList/MoodOptionList";
+import Stepper from "../Stepper/Stepper";
 
 type LogMoodModalProps = {
   closeLog: () => void;
 };
 
 const LogMoodModal = ({ closeLog }: LogMoodModalProps) => {
   const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const [step, setStep] = useState<number>(0);
+
+  const modalRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        modalRef.current &&
+        !modalRef.current.contains(event.target as Node)
+      ) {
+        closeLog();
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
```

Comment:
Using ref should prevent you from having to touch the dom. Avoid this, and try to recreate this solution